tests/baselines/reference/chainedCallsWithTypeParameterConstrainedToOtherTypeParameter.types

```diff
@@ -79,7 +79,7 @@ class C extends B {
 >A : typeof A
 >  : ^^^^^^^^
 >then : <S extends A>(cb: (x: A) => S) => Chain<S>

```

Comment:
In examples like these, I think it's incredibly weird that the old behavior had us not reusing effectively anything _but_ the return type annotation.